### PR TITLE
unix: remove clang compiler warning pragmas

### PR DESCRIPTION
**Pull request by bnoordhuis**
Instead of suppressing the VLA warning, use a fixed-size buffer that is big enough to receive at least one control message but not so big that IBMi PASE rejects it.

### CODE REVIEW

File: src/unix/stream.c [hunk 1]
```diff
@@ -1037,17 +1028,11 @@ static int uv__stream_recv_cmsg(uv_stream_t* stream, struct msghdr* msg) {
...
-# pragma clang diagnostic ignored "-Wgnu-folding-constant"
-# pragma clang diagnostic ignored "-Wvla-extension"
-#endif
-
 static void uv__read(uv_stream_t* stream) {
   uv_buf_t buf;
   ssize_t nread;
   struct msghdr msg;
-  char cmsg_space[CMSG_SPACE(UV__CMSG_FD_SIZE)];
+  char cmsg_space[256];
```
Comment (vtjnash) [inline src/unix/stream.c:1035]: This seems probably too large (https://github.com/libuv/libuv/pull/2591)

Comment (bnoordhuis) [reply src/unix/stream.c]: I've updated it to reuse the union that we're already using elsewhere, PTAL.

File: src/unix/stream.c [hunk 2]
```diff
@@ -769,18 +779,14 @@ static int uv__try_write(uv_stream_t* stream,
...
-    } scratch;
+    union uv__cmsg cmsg;
 
     if (uv__is_closing(send_handle))
       return UV_EBADF;
 
     fd_to_send = uv__handle_fd((uv_handle_t*) send_handle);
 
-    memset(&scratch, 0, sizeof(scratch));
+    memset(&cmsg, 0, sizeof(cmsg));
```
Comment (vtjnash) [inline src/unix/stream.c:789]: small nit: I think C++ compilers might have rejected this, though it was valid in C.
```suggestion
    memset(&cmsg.pad, 0, sizeof(cmsg));
```

**Review by vtjnash — COMMENTED** [review]
Yes, this approach seems cleaner to me also

File: src/unix/stream.c (hunk 2 above)
Comment (bnoordhuis) [reply src/unix/stream.c:789]: The past tense confuses me. It _is_ valid C, right? :-)

**Review by santigimeno — APPROVED** [review]

**Pull request merged**